src/impl.ml

```diff
@@ -108,7 +108,8 @@ let contents common filename =
         end;
         tl () >>= fun x ->
         loop x in
-      openstream (Input.of_fd (Vhd_format_lwt.IO.to_file_descr fd)) >>= fun stream ->
+      Vhd_format_lwt.IO.get_file_size filename >>= fun size ->
```

Comment:
Could openfile+get_file_size+openstream go into a new convenience function? `fd` itself doesn't seem to be needed.